xonsh/foreign_shells.py

```diff
@@ -80,62 +82,65 @@
 echo ${namefile}
 """.strip()
 
+# mapping of shell name alises to keys in other lookup dictionaries.
+CANON_SHELL_NAMES = {
+    'bash': 'bash',
+    '/bin/bash': 'bash',
+    'zsh': 'zsh',
+    '/bin/zsh': 'zsh',
+    '/usr/bin/zsh': 'zsh',
+    'cmd': 'cmd',
+    'cmd.exe': 'cmd',
+}
+
 DEFAULT_ENVCMDS = {
     'bash': 'env',
-    '/bin/bash': 'env',
     'zsh': 'env',
-    '/bin/zsh': 'env',
-    '/usr/bin/zsh': 'env',
     'cmd': 'set',
 }
 DEFAULT_ALIASCMDS = {
     'bash': 'alias',
-    '/bin/bash': 'alias',
     'zsh': 'alias -L',
-    '/bin/zsh': 'alias -L',
-    '/usr/bin/zsh': 'alias -L',
     'cmd': '',
 }
 DEFAULT_FUNCSCMDS = {
     'bash': DEFAULT_BASH_FUNCSCMD,
-    '/bin/bash': DEFAULT_BASH_FUNCSCMD,
     'zsh': DEFAULT_ZSH_FUNCSCMD,
-    '/bin/zsh': DEFAULT_ZSH_FUNCSCMD,
-    '/usr/bin/zsh': DEFAULT_ZSH_FUNCSCMD,
     'cmd': '',
 }
 DEFAULT_SOURCERS = {
     'bash': 'source',
-    '/bin/bash': 'source',
     'zsh': 'source',
-    '/bin/zsh': 'source',
-    '/usr/bin/zsh': 'source',
     'cmd': 'call',
 }
 DEFAULT_TMPFILE_EXT = {
     'bash': '.sh',
-    '/bin/bash': '.sh',
     'zsh': '.zsh',
-    '/bin/zsh': '.zsh',
-    '/usr/bin/zsh': '.zsh',
     'cmd': '.bat',
 }
 DEFAULT_RUNCMD = {
     'bash': '-c',
-    '/bin/bash': '-c',
     'zsh': '-c',
-    '/bin/zsh': '-c',
-    '/usr/bin/zsh': '-c',
     'cmd': '/C',
 }
+DEFAULT_SETERRPREVCMD = {
+    'bash': 'set -e',
+    'zsh': 'set -e',
+    'cmd': '',
```

Comment:
On windows this could default to `@echo off`. If `echo off` is not set all the  that is not set all the `__XONSH_XXX_BEG__` tokens are repeated twice in the output